### DISCUSSION TITLE
Why doesn't boost/geometry support wkb io?

### DISCUSSION THREAD

**Discussion by jievince** [boostorg/geometry#912] [Q&A]
It's very important.

**Comment by mloskot** [comment]
https://github.com/boostorg/geometry/search?q=wkb&type=code

**Comment by awulkiew** [comment]
@jievince WKB is supported as an extension (so not officially released). The fact that it's in the extensions probably means that it's not ready to be released, but I don't really know. You can use it after getting it from the develop branch.

@meastp was working on it (https://github.com/boostorg/geometry/pull/145). If it's important you could consider continuing his work.

**Reply by meastp** [reply]
I consider the work I did as finished. as we've been using wkb io in production for many years now. @awulkiew

**Reply by awulkiew** [reply]
Ok thanks! I skimmed through the code and I don't think that it's ready to be released. Unless I'm missing something. In this case please correct me. So AFAIU someone has to take it from here because:

- Currently only point, linestring and polygon are supported, at least all OGC geometries should've been and maybe other BG non-standard ones but that's debatable.
- Looking at the code I'm also not sure if the handling of dimensions is correct. It is possible to pass 1D or 4D geometries right now which would compile and any errors wouldn't be reported. In WKB it is not possible to define them. We could think about our own extensions of the standard but IDK. AFAIU right now the dimensionality is defined by the C++ type of the geometry and this has to match the data in WKB. If a user makes a mistake here there is no mechanism of reporting it.

**Reply by meastp** [reply]
@awulkiew At least Multi* -geometries are also supported, but they are in the `multi` folder instead :)

**Reply by awulkiew** [reply]
@meastp Ah ok, thanks. I forgot that we used this directory structure in the past. So OGC geometries are supported then.

And the tests are there too. Though AFAIS unexpected things like not matching dimensions are not tested.

**Reply by barendgehrels** [reply]
I'm in favor of moving WKB to our mainline (but I currently cannot spend time on it). Good to hear that it's in production for many years!
